patches/tModLoader/Terraria/release_extras/start-tModLoaderServer.sh

```diff
@@ -1,23 +1,58 @@
 #!/usr/bin/env bash
-cd "$(dirname "$0")"
+cd "$(dirname "$0")" || exit
 
-launch_args="$@ -server"
-if [[ ! "$launch_args" == *"-config"* ]]; then
+launch_args="-server"
+
+# Parse parameters, passing through everything else to ScriptCaller.sh
+while [[ $# -gt 0 ]]
+do
+	case $1 in
+		-steam)
+			steam_server=true
+			shift
+			;;
+		-nosteam)
+			steam_server=false
+			shift
+			;;
+		*) # If it's not one of the args above, then pass it to ScriptCaller.sh
+			launch_args="$launch_args $1"
+			shift
+			;;
+	esac
+done
+
+# Use serverconfig.txt as config if not already specified
+if ! [[ "$launch_args" == *"-config"* ]]
+then
```

Comment:
Code style, can we put `then` on the same line as the `if`? Is there a reason this is inadvisable in bash?

Reply:
No reason not to, I just thought it looked nicer this way. Easy change though.